community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/TransactionMetadataCache.java

```diff
@@ -124,37 +135,18 @@ public boolean equals( Object o )
             {
                 return false;
             }
-
             TransactionMetadata that = (TransactionMetadata) o;
-
-            if ( authorId != that.authorId )
-            {
-                return false;
-            }
-            if ( checksum != that.checksum )
-            {
-                return false;
-            }
-            if ( masterId != that.masterId )
-            {
-                return false;
-            }
-            if ( !startPosition.equals( that.startPosition ) )
-            {
-                return false;
-            }
-
-            return true;
+            return masterId == that.masterId &&
+                   authorId == that.authorId &&
+                   checksum == that.checksum &&
+                   timeWritten == that.timeWritten &&
+                   Objects.equals( startPosition, that.startPosition );
         }
 
         @Override
         public int hashCode()
         {
-            int result = masterId;
-            result = 31 * result + authorId;
-            result = 31 * result + startPosition.hashCode();
-            result = 31 * result + (int) (checksum ^ (checksum >>> 32));
-            return result;
+            return Objects.hash( masterId, authorId, startPosition, checksum, timeWritten );
```

Comment:
Looks like `equals` and `hashCode` are not strictly necessary on `TransactionMetadata` because they are never used as keys in hashed collections, or otherwise used in comparisons. Also, this mechanism of computing an objects hash code causes allocations – boxing for primitive elements, and then the carrier array itself – so I'd discourage it.